linkerd/protocol/http/src/main/scala/io/buoyant/linkerd/protocol/http/ResponseClassifiers.scala

```diff
@@ -58,14 +60,32 @@ object ResponseClassifiers {
     }
   }
 
+  object RetryableException {
+    def unapply(e: Throwable): Boolean = e match {
+      case RetryableWriteException(_) => true
+      case Failure(Some(_: TimeoutException)) => true
+      case Failure(Some(_: UtilTimeoutException)) => true
+      case _: TimeoutException => true
+      case _: UtilTimeoutException => true
+      case _: ChannelClosedException => true
+      case _ => false
+    }
```

Comment:
this signatures of finagle's things are a bit funky.. i'll see if i can reuse it, though